### PR TITLE
Fix docker

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Docker file was broken, so now added a workflow that builds docker file on every push

## How Has This Been Tested?

CI workflow builds the docker file

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Build docker
+
+on:
+  push:
+  schedule:
+    - cron: "0 12 * * 1"
```
Comment (simlmx) [inline .github/workflows/build-docker.yml:6]: What's the reason to build on a cron?

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  push_to_registry:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        with:
+          submodules: true
```
Comment (simlmx) [inline .github/workflows/build-docker.yml:17]: Does the repo have submodules?

File: infrastructure/docker/Dockerfile [hunk 3]
```diff
@@ -18,15 +18,15 @@ COPY pyproject.toml app/pyproject.toml
...
-RUN poetry install --witout dev --no-cache
-
 # copy library files
 COPY pvconsumer/ app/pvconsumer/
 
 # change to app folder
 WORKDIR /app
 
+# Install python requirements.
+RUN poetry install --no-cache
```
Comment (simlmx) [inline infrastructure/docker/Dockerfile:28]: Why remove the `--without dev`?

File: .github/workflows/build-docker.yml (hunk 1 above)
Comment (peterdudfield) [reply .github/workflows/build-docker.yml:6]: so we know it always works

File: .github/workflows/build-docker.yml (hunk 2 above)
Comment (peterdudfield) [reply .github/workflows/build-docker.yml:17]: nope, should remove

File: .github/workflows/build-docker.yml (hunk 1 above)
Comment (simlmx) [reply .github/workflows/build-docker.yml:6]: Anything that *can* be different between two runs if the code doesn't change?

**Pull request merged**